src/curl.c

```diff
@@ -59,6 +59,8 @@ curl_multi_process(CURLM *multi, curl_socket_t sockfd, int events)
 	 * more data to send off before it is "satisfied".
 	 */
 	do {
+		if (!multi)
+			return;
```

Comment:
> Moved to the top of the function.

Somewhy I don't see this change. We don't need to check this value for each iteration. It's enough to check it only once

Reply:
> It doesn't set env->multi to NULL. It just deallocates a memory. After that pointer still points to some memory (that you doesn't own anymore).

1. `curl_multi_cleanup`, https://github.com/curl/curl/blob/ca6bafce95365fadd906a533bb16e47b638c5c2a/lib/multi.c#L2748C11-L2748C29
2. `curl_multi_remove_handle`, https://github.com/curl/curl/blob/ca6bafce95365fadd906a533bb16e47b638c5c2a/lib/multi.c#L930

Reply:
My question is not about "it's broken". My question is do we really need this condition and that will happen in case if it will be removed?

Reply:
> My question is do we really need this condition and that will happen in case if it will be removed?

Initially, it was added as a protection for a case when env is destroyed and we have an active ev callback with http request. I've removed condition and checked on etcd-client tests, seems this condition is not required anymore.